Include 3rdparty directory into the main build-system.

Original author is unknown.
(Was either Juan Quintela or Jeff Garzik)

Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
Signed-off-by: Herton Ronaldo Krzesinski <herton@mandriva.com.br>
---
 Makefile                 |    2 +-
 arch/alpha/Kconfig       |    1 +
 arch/ia64/Kconfig        |    2 ++
 arch/mips/Kconfig        |    2 ++
 arch/powerpc/Kconfig     |    2 ++
 arch/sparc/Kconfig       |    2 ++
 arch/x86/Kconfig         |    2 ++
 scripts/kconfig/Makefile |   31 +++++++++++++++++--------------
 8 files changed, 29 insertions(+), 15 deletions(-)

diff -Nurp linux-3.1-rc9/arch/alpha/Kconfig linux-3.1-rc9-3rd/arch/alpha/Kconfig
--- linux-3.1-rc9/arch/alpha/Kconfig	2011-10-05 19:52:41.318669657 +0300
+++ linux-3.1-rc9-3rd/arch/alpha/Kconfig	2011-10-05 20:54:02.210746844 +0300
@@ -676,3 +676,4 @@ source "crypto/Kconfig"
 
 source "lib/Kconfig"
 
+source "3rdparty/Kconfig"
diff -Nurp linux-3.1-rc9/arch/ia64/Kconfig linux-3.1-rc9-3rd/arch/ia64/Kconfig
--- linux-3.1-rc9/arch/ia64/Kconfig	2011-10-05 19:52:41.617671407 +0300
+++ linux-3.1-rc9-3rd/arch/ia64/Kconfig	2011-10-05 20:56:11.469682272 +0300
@@ -667,3 +667,5 @@ source "lib/Kconfig"
 
 config IOMMU_HELPER
 	def_bool (IA64_HP_ZX1 || IA64_HP_ZX1_SWIOTLB || IA64_GENERIC || SWIOTLB)
+
+source "3rdparty/Kconfig"
diff -Nurp linux-3.1-rc9/arch/mips/Kconfig linux-3.1-rc9-3rd/arch/mips/Kconfig
--- linux-3.1-rc9/arch/mips/Kconfig	2011-10-05 19:52:41.639671536 +0300
+++ linux-3.1-rc9-3rd/arch/mips/Kconfig	2011-10-05 20:54:02.210746844 +0300
@@ -2494,3 +2494,5 @@ source "security/Kconfig"
 source "crypto/Kconfig"
 
 source "lib/Kconfig"
+
+source "3rdparty/Kconfig"
diff -Nurp linux-3.1-rc9/arch/powerpc/Kconfig linux-3.1-rc9-3rd/arch/powerpc/Kconfig
--- linux-3.1-rc9/arch/powerpc/Kconfig	2011-10-05 19:52:41.674671739 +0300
+++ linux-3.1-rc9-3rd/arch/powerpc/Kconfig	2011-10-05 20:54:02.211746759 +0300
@@ -999,3 +999,5 @@ config PPC_LIB_RHEAP
 	bool
 
 source "arch/powerpc/kvm/Kconfig"
+
+source "3rdparty/Kconfig"
diff -Nurp linux-3.1-rc9/arch/sparc/Kconfig linux-3.1-rc9-3rd/arch/sparc/Kconfig
--- linux-3.1-rc9/arch/sparc/Kconfig	2011-10-05 19:52:41.766672278 +0300
+++ linux-3.1-rc9-3rd/arch/sparc/Kconfig	2011-10-05 20:54:02.211746759 +0300
@@ -605,3 +605,5 @@ source "security/Kconfig"
 source "crypto/Kconfig"
 
 source "lib/Kconfig"
+
+source "3rdparty/Kconfig"
diff -Nurp linux-3.1-rc9/arch/x86/Kconfig linux-3.1-rc9-3rd/arch/x86/Kconfig
--- linux-3.1-rc9/arch/x86/Kconfig	2011-10-05 19:52:41.800672479 +0300
+++ linux-3.1-rc9-3rd/arch/x86/Kconfig	2011-10-05 20:54:02.211746759 +0300
@@ -2151,3 +2151,5 @@ source "crypto/Kconfig"
 source "arch/x86/kvm/Kconfig"
 
 source "lib/Kconfig"
+
+source "3rdparty/Kconfig"
diff -Nurp linux-3.1-rc9/Makefile linux-3.1-rc9-3rd/Makefile
--- linux-3.1-rc9/Makefile	2011-10-05 19:52:41.318669657 +0300
+++ linux-3.1-rc9-3rd/Makefile	2011-10-05 20:54:02.211746759 +0300
@@ -507,7 +507,7 @@ scripts: scripts_basic include/config/au
 
 # Objects we will link into vmlinux / subdirs we need to visit
 init-y		:= init/
-drivers-y	:= drivers/ sound/ firmware/
+drivers-y	:= drivers/ sound/ firmware/ 3rdparty/
 net-y		:= net/
 libs-y		:= lib/
 core-y		:= usr/
--- linux/scripts/kconfig/Makefile.3rd-3rdparty-merge.orig	2011-12-24 13:13:38.396402649 +0200
+++ linux/scripts/kconfig/Makefile	2011-12-24 22:01:23.383239314 +0200
@@ -11,29 +11,29 @@ else
 Kconfig := Kconfig
 endif
 
-xconfig: $(obj)/qconf
+xconfig: $(obj)/qconf 3rdparty/Makefile
 	$< $(Kconfig)
 
-gconfig: $(obj)/gconf
+gconfig: $(obj)/gconf 3rdparty/Makefile
 	$< $(Kconfig)
 
-menuconfig: $(obj)/mconf
+menuconfig: $(obj)/mconf 3rdparty/Makefile
 	$< $(Kconfig)
 
-config: $(obj)/conf
+config: $(obj)/conf 3rdparty/Makefile
 	$< --oldaskconfig $(Kconfig)
 
-nconfig: $(obj)/nconf
+nconfig: $(obj)/nconf 3rdparty/Makefile
 	$< $(Kconfig)
 
-oldconfig: $(obj)/conf
+oldconfig: $(obj)/conf 3rdparty/Makefile
 	$< --$@ $(Kconfig)
 
-silentoldconfig: $(obj)/conf
+silentoldconfig: $(obj)/conf 3rdparty/Makefile
 	$(Q)mkdir -p include/generated
 	$< --$@ $(Kconfig)
 
-localyesconfig localmodconfig: $(obj)/streamline_config.pl $(obj)/conf
+localyesconfig localmodconfig: $(obj)/streamline_config.pl $(obj)/conf 3rdparty/Makefile
 	$(Q)mkdir -p include/generated
 	$(Q)perl $< --$@ $(srctree) $(Kconfig) > .tmp.config
 	$(Q)if [ -f .config ]; then 					\
@@ -73,18 +73,18 @@ update-po-config: $(obj)/kxgettext $(obj
 
 PHONY += allnoconfig allyesconfig allmodconfig alldefconfig randconfig
 
-allnoconfig allyesconfig allmodconfig alldefconfig randconfig: $(obj)/conf
+allnoconfig allyesconfig allmodconfig alldefconfig randconfig: $(obj)/conf 3rdparty/Makefile
 	$< --$@ $(Kconfig)
 
 PHONY += listnewconfig oldnoconfig savedefconfig defconfig
 
-listnewconfig oldnoconfig: $(obj)/conf
+listnewconfig oldnoconfig: $(obj)/conf 3rdparty/Makefile
 	$< --$@ $(Kconfig)
 
-savedefconfig: $(obj)/conf
+savedefconfig: $(obj)/conf 3rdparty/Makefile
 	$< --$@=defconfig $(Kconfig)
 
-defconfig: $(obj)/conf
+defconfig: $(obj)/conf 3rdparty/Makefile
 ifeq ($(KBUILD_DEFCONFIG),)
 	$< --defconfig $(Kconfig)
 else
@@ -92,7 +92,7 @@ else
 	$(Q)$< --defconfig=arch/$(SRCARCH)/configs/$(KBUILD_DEFCONFIG) $(Kconfig)
 endif
 
-%_defconfig: $(obj)/conf
+%_defconfig: $(obj)/conf 3rdparty/Makefile
 	$(Q)$< --defconfig=arch/$(SRCARCH)/configs/$@ $(Kconfig)
 
 # Help text used by make help
@@ -169,6 +169,8 @@ ifeq ($(MAKECMDGOALS),gconfig)
 	gconf-target := 1
 endif
 
+3rdparty/Makefile:
+	pushd $(srctree)/3rdparty ; $(PERL) ./mkbuild.pl ; popd
 
 ifeq ($(qconf-target),1)
 	hostprogs-y += qconf
